File: python/util/hash.py
import hashlib, hmac, binascii

# 哈希
class Hash:

  # Md5
  def Md5(data: str):
    return hashlib.md5(data.encode(encoding='utf-8')).hexdigest()

  # Sha256
  def Sha256(data: str):
    return hashlib.sha256(data.encode(encoding='utf-8')).hexdigest()

  # HmacSha1
  def HmacSha1(data: str, key: bytes):
    return hmac.new(key, str.encode(data), hashlib.sha1).digest()

  # HmacSha256
  def HmacSha256(data: str, key: bytes):
    return hmac.new(key, str.encode(data), hashlib.sha256).digest()

  # Bytes转为16进制
  def HexEncode(byte: bytes):
    return binascii.b2a_hex(byte).decode('utf-8')

File: php/util/Hash.php
<?php
namespace Util;

/* 哈希 */
class Hash {

  /* Md5 */
  static function Md5(string $data, $binary=true): string {
    return md5($data, $binary);
  }

  /* Sha256 */
  static function Sha256(string $data): string {
    $res = hash('SHA256', $data);
    return $res!==false?$res:'';
  }

  /* HmacSha1 */
  static function HmacSha1(string $data, string $key, bool $binary=true): string {
    $res = hash_hmac('sha1', $data, $key, $binary);
    return $res!==false?$res:'';
  }

  /* HmacSha256 */
  static function HmacSha256(string $data, string $key, bool $binary=true): string {
    $res = hash_hmac('sha256', $data, $key, $binary);
    return $res!==false?$res:'';
  }

  /* GetBytes */
  static function GetBytes(string $data): array {
    $bytes = [];
    for($i = 0; $i < strlen($data); $i++){
      $bytes[] = ord($data[$i]);
    }
    return $bytes;
  }

}